src/coffea/dataset_tools/preprocess.py

```diff
@@ -227,11 +231,14 @@ def _normalize_file_info(file_info):
             None if not isinstance(maybe_finfo, dict) else maybe_finfo.get("uuid", None)
         )
         this_file = normed_files[ifile]
-        this_file += (3 - len(this_file)) * (None,) + (maybe_uuid,)
+        this_file += (4 - len(this_file)) * (None,) + (maybe_uuid,)
```

Comment:
🤨

Reply:
I added `num_entries` as a field. See later.

Reply:
This would be a good candidate to turn into a dataclass (or pydantic model maybe? since you want to be able to handle awkward arrays of them)

Reply:
Probably but I'm finding little motivation for the additional work? Certainly seems correct but low priority.